rwkv.cpp

```diff
@@ -600,6 +605,45 @@ struct rwkv_context * rwkv_init_from_file(const char * file_path, const uint32_t
         RWKV_ASSERT_NULL(RWKV_ERROR_MODEL_PARAMS, set_block_parameter(&parameters, i, "ffn.receptance.weight", &layer->ffn_receptance));
     }
 
+#ifdef GGML_USE_CUBLAS
+    {
+        const int n_gpu = std::min(n_gpu_layers, model->n_layer);
+
+        fprintf(stdout, "%s: [cuBLAS] offloading %d layers to GPU\n", __func__, n_gpu);
+
+        size_t vram_total = 0;
+
+        for (int i = 0; i < n_gpu; ++i) {
+            const auto & layer = model->layers[i];
+
+            ggml_cuda_transform_tensor(layer.ln1_weight); vram_total += ggml_nbytes(layer.ln1_weight);
```

Comment:
LN weights are tiny and have the same size as LN biases, I'm 95% sure they should not be on GPU. This will not save much VRAM, but will simplify code

Reply:
multiplies with these tensors will not be optimized if they are not on the GPU though

Reply:
But after multiplying by `ln.weight` `x` then is added to `ln.bias`, which will be on CPU -- so we will still need to download `x` from the GPU, in both cases. And multiplying 1.5K floats (as I remember, largest `n_embed` for RWKV) looks very okay for CPU.

Reply:
maybe you're right, I did not look at the surrounding context

Reply:
If we reallly want to get rid of CPU-GPU transfer in first layers, then all tensors of first layers (including biases) needs to be on GPU.

Unfortunately, as I remember, ggml does not support any operations on GPU aside from matmul.

Reply:
Somehow, by removing `ln.weight` the inference speed get increased by 10 - 20%